File: Optica.Api/Modules/Clientes/Models/Cliente.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Optica.Api.Modules.Clientes.Models;

[Table("clientes")]
public class Cliente
{
    [Key]
    [Column("id_cliente")]
    public int IdCliente { get; set; }

    [Column("rut")]
    [Required]
    [MaxLength(12)]
    public string Rut { get; set; } = string.Empty;

    [Column("nombre")]
    [Required]
    [MaxLength(60)]
    public string Nombre { get; set; } = string.Empty;

    [Column("apellido")]
    [Required]
    [MaxLength(60)]
    public string Apellido { get; set; } = string.Empty;

    [Column("telefono")]
    [MaxLength(20)]
    public string? Telefono { get; set; }

    [Column("correo")]
    [MaxLength(100)]
    public string? Correo { get; set; }

    [Column("estado")]
    public string Estado { get; set; } = "Activo";

    [Column("fecha_registro")]
    public DateTime FechaRegistro { get; set; } = DateTime.Now;
}

File: Optica_front/src/app/services/cliente.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

export interface Cliente {
  idCliente?: number;
  rut: string;
  nombre: string;
  apellido: string;
  telefono?: string;
  correo?: string;
  estado?: string;
  fechaRegistro?: string;
}

@Injectable({
  providedIn: 'root'
})
export class ClienteService {
  private apiUrl = 'http://localhost:8080/api/Clientes';

  constructor(private http: HttpClient) {}

  registrar(cliente: Cliente): Observable<Cliente> {
    return this.http.post<Cliente>(this.apiUrl, cliente);
  }

  buscar(termino: string = ''): Observable<Cliente[]> {
    let params = new HttpParams();
    if (termino.trim()) {
      params = params.set('termino', termino.trim());
    }
    return this.http.get<Cliente[]>(`${this.apiUrl}/buscar`, { params });
  }
}